src/indices/indexed-class.lisp

```diff
@@ -293,31 +296,53 @@ also index subclasses of the class to which the slot belongs, default is T")
 
 (defvar *indexed-class-override* nil)
 
-(defmethod slot-value-using-class :before ((class indexed-class) object slot)
+(defmethod slot-value-using-class :before ((class indexed-class) object
+                                           (slot slot-definition))
   (when (and (not (eql (slot-definition-name slot) 'destroyed-p))
 	     (object-destroyed-p object)
 	     (not *indexed-class-override*))
     (error "Can not get slot ~A of destroyed object of class ~a."
-	   (slot-definition-name slot) (class-name (class-of object)))))
+	       (slot-definition-name slot) (class-name (class-of object)))))
+
+#+lispworks
+(defmethod slot-value-using-class ((class indexed-class) object
+                                   (slot symbol))
+  (slot-value-using-class class
+                          object
+                          (clos:find-slot-definition slot class)))
```

Comment:
here's an example of the specializer. It needs to be careful though: I couldn't specialize on the :after/:before methods directly, because then (call-next-methods) become really tricky.